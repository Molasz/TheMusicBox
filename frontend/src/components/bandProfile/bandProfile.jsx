import React, { useEffect } from 'react';
import { connect } from 'react-redux';

import './bandProfile.scss';

import { getBand } from '../../redux/actions/bandActions';
import store from '../../redux/store';

import Photos from './photos/photos';
import Discography from './discography/discography';
import Concerts from './concerts/concerts';
import Bio from './bio/bio';

import GradeIcon from '@material-ui/icons/Grade';

function BandProfile(props) {
  useEffect(() => {
    store.dispatch(getBand(props.match.params.bandId));
  }, []);

  const result = props.band ? (
    <article className='band-profile'>
      <div className='band-profile__top'>
        <img src={props.band.logo} alt='Logo' className='top__logo' />

        <img
          src='https://pbs.twimg.com/profile_banners/851494228815482880/1539011178/1500x500'
          alt='Banner'
          className='top__banner'
        />
        <div className='top__info'>
          <strong className='info__name'>{props.band.name}</strong>
          <div className='info__follow'>
            <GradeIcon />
            <button className='follow__button'>Follow</button>
            <p className='follow__number'> 14 Followers</p>
          </div>
        </div>
      </div>
      <div className='band-profile__middle'>
        <div className='middle__bio'>
          <Bio
            bio={props.band.bio}
            city={props.band.city}
            country={props.band.country}
            name={props.band.name}
          />
        </div>

        <Discography data={props.band.discography} />
      </div>
      <div className='band-profile__bottom'>
        <div className='bottom__concerts'>
          <Concerts data={props.band.concerts} />
        </div>
        <div className='bottom__photos'>
          <Photos data={props.band.photos} />
        </div>
      </div>
    </article>
  ) : (
    <h3>Loading...</h3>
  );
  return result;
}

function mapStateToProps(state) {
  return state.bandReducer;
}

function mapDispatchToProps(dispatch) {
  return { getBand };
}
export default connect(mapStateToProps, mapDispatchToProps)(BandProfile);
