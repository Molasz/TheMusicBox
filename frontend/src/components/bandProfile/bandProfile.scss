$color-detail: #f55110;
.band-profile {
  min-width: 500px;
  display: flex;
  flex-direction: column;

  color: white;
  padding: 20px 10%;
  .band-profile__top {
    display: flex;
    flex-direction: row;

    padding-bottom: 13px;
    border-bottom: 4px solid $color-detail;
    margin-bottom: 0px;
    .top__logo {
      width: 50%;
    }
    .top__banner {
      display: none;
      width: 50%;

      margin-left: 30px;
    }
    .top__info {
      display: flex;
      flex-direction: column;
      justify-content: space-between;

      width: 30%;
      margin-left: 30px;

      .info__name {
        text-transform: uppercase;
        font-size: 80px;
        margin: 0;
      }

      .info__follow {
        .follow__container {
          display: flex;
          align-items: center;

          .contanier__icon {
            font-size: 30px;
            margin-right: 10px;
          }

          .white {
            color: white;
          }

          .orange {
            color: $color-detail;
          }
        }
      }
    }
  }
  .band-profile__middle {
    display: flex;
    flex-direction: column;

    padding-bottom: 20px;
    border-bottom: 4px solid $color-detail;
  }
  .band-profile__bottom {
    display: flex;
    flex-direction: column;
  }
}

// Medium devices (tablets, 768px and up)
@media (min-width: 768px) {
}

// Large devices (desktops, 992px and up)
@media (min-width: 992px) {
  .band-profile {
    .band-profile__top {
      .top__logo {
        width: 20%;
      }
      .top__banner {
        display: block;
      }
      .top__info {
        display: flex;
        flex-direction: column;

        margin-left: 30px;
        .info__name {
          font-size: 80px;
        }
      }
    }
    .band-profile__middle {
      flex-direction: row;
    }

    .band-profile__bottom {
      flex-direction: row;
    }
  }
}
