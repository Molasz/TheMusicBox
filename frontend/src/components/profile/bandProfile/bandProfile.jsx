import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';

import './bandProfile.scss';

import PuffLoader from 'react-spinners/PuffLoader';

import { getBand, follow, showDisc } from '../../../redux/actions/bandActions';

import ProfileHeader from '../profileHeader/profileHeader';
import Photos from './photos/photos';
import Discography from './discography/discography';
import Concerts from './concerts/concerts';
import Bio from '../bio/bio';

function BandProfile({ match, band, followers, user, dispatch }) {
  const [calls, setCalls] = useState(false);
  const [checkFollowing, setCheckFollowing] = useState(false);
  const [isFollowing, setIsFollowing] = useState(null);

  useEffect(() => {
    if (!calls) {
      dispatch(showDisc());
      dispatch(getBand(match.params.bandId));
      dispatch(follow(match.params.bandId));
      setCalls(true);
    }
    if (
      !checkFollowing &&
      !(Object.keys(user).length === 0 && user.constructor === Object) &&
      !(Object.keys(band).length === 0 && band.constructor === Object)
    ) {
      setIsFollowing(
        user.following.some((element) => element._id === band._id)
      );
      setCheckFollowing(true);
    }
  }, [band, calls, dispatch, match.params.bandId, user, checkFollowing]);

  const result = !(
    Object.keys(band).length === 0 && band.constructor === Object
  ) ? (
    <article className='band-profile'>
      <div className='band-profile__top'>
        <ProfileHeader
          logo={band.logo}
          banner={band.banner}
          name={band.name}
          isBand={true}
          match={match}
        />
      </div>
      <div className='band-profile__middle'>
        <Bio
          bio={band.bio}
          city={band.city}
          country={band.country}
          name={band.name}
          socialNetwork={band.socialNetwork}
        />

        <Discography data={band.discography} />
      </div>
      <div className='band-profile__bottom'>
        <Concerts data={band.concerts} />
        <Photos data={band.photos} />
      </div>
    </article>
  ) : (
    <PuffLoader color='#f55110' />
  );

  return result;
}

function mapStateToProps(state) {
  return {
    band: state.bandReducer.band,
    user: state.authReducer.user
  };
}

export default connect(mapStateToProps, null)(BandProfile);
export { BandProfile };
