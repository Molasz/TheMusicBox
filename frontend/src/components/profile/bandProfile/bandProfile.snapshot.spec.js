import React from 'react';
import renderer from 'react-test-renderer';

import { BandProfile } from './bandProfile';

import BandBio from './bandBio/bandBio';
jest.mock('./bandBio/bandBio');
BandBio.mockReturnValue(<></>);

jest.mock('./discography/discography');
jest.mock('./bandHeader/bandHeader');
//jest.mock('./photos/photos');

describe('BandProfile snapshot', () => {
  const match = { params: { bandId: 1 } };
  const band = {
    socialNetwork: ['twitter'],
    photos: ['photo'],
    _id: 1,
    name: 'name',
    city: 'city',
    country: 'country',
    tags: ['tag'],
    bio: 'bio',
    logo: 'logo',
    banner: 'banner',
    discography: [],
    concerts: [{ date: 'date', city: 'city' }]
  };
  const followers = 1312;
  const user = { following: ['', ''] };
  const dispatch = jest.fn();

  it('Should match with data', () => {
    const tree = renderer.create(
      <BandProfile
        match={match}
        band={band}
        user={user}
        followers={followers}
        dispatch={dispatch}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should match without data', () => {
    const tree = renderer.create(
      <BandProfile match={match} band={{}} dispatch={dispatch} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
