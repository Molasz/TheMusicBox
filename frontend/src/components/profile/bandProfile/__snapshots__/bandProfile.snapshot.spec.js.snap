// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandProfile snapshot Should match with data 1`] = `
<article
  className="band-profile"
>
  <div
    className="band-profile__middle"
  />
  <div
    className="band-profile__bottom"
  >
    <section
      className="concerts"
    >
      <div
        className="concerts__title"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root title__icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
          />
        </svg>
        <h1
          className="title__text"
        >
          Concerts
        </h1>
      </div>
      <div
        className="concerts__main"
      >
        <div
          className="main__item"
        >
          <p
            className="item__date"
          >
            date
          </p>
          <p
            className="item__city"
          >
            city
          </p>
        </div>
      </div>
    </section>
    <section
      className="photo"
    >
      <div
        className="photo__title"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root title__icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
          />
        </svg>
        <h1
          className="title__text"
        >
          Photos and videos
        </h1>
      </div>
      <div
        className="photo__main"
      >
        <img
          alt="nÂº0"
          className="main__item"
          src="photo"
        />
      </div>
    </section>
  </div>
</article>
`;

exports[`BandProfile snapshot Should match without data 1`] = `
<div
  className="css-3fvcat"
>
  <div
    className="css-hwvl8z"
  />
  <div
    className="css-16hco3u"
  />
</div>
`;
